sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -363,11 +363,32 @@ static bool _pio_from_iface_addr(gnrc_pktsnip_t **res, gnrc_ipv6_netif_addr_t *a
     return false;
 }
 
+static inline bool _check_prefixes(gnrc_ipv6_netif_addr_t *a, gnrc_ipv6_netif_addr_t *b)
+{
+    if ((a->prefix_len < 128U) && (a->prefix_len == b->prefix_len) &&
+        (ipv6_addr_match_prefix(&a->addr, &b->addr) >= a->prefix_len)) {
+        return true;
+    }
+
+    return false;
+}
+
 static gnrc_pktsnip_t *_add_pios(gnrc_ipv6_netif_t *ipv6_iface, gnrc_pktsnip_t *pkt)
 {
     gnrc_pktsnip_t *tmp;
+    bool processed_before = false;
+
     for (int i = 0; i < GNRC_IPV6_NETIF_ADDR_NUMOF; i++) {
-        if (_pio_from_iface_addr(&tmp, &ipv6_iface->addrs[i], pkt)) {
+        /* skip if prefix has been processed already */
+        processed_before = false;
+        for (int j = 0; j < i; j++) {
+            if ((processed_before =
+                _check_prefixes(&ipv6_iface->addrs[i], &ipv6_iface->addrs[j]))) {
+                break;
+            }
+        }
+
```

Comment:
do you mean instead of:
```c
if ((!processed_before) && _pio_from_iface_addr(&tmp, &ipv6_iface->addrs[i], pkt)) {
```
you would prefer:
```c
if (processed_before) {
    continue;
}

if (_pio_from_iface_addr(&tmp, &ipv6_iface->addrs[i], pkt)) {
```

?